simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -921,6 +921,14 @@ class const_where_expression<
     _mm256_mask_storeu_epi32(mem, static_cast<__mmask8>(m_mask),
                              static_cast<__m256i>(m_value));
   }
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+  void scatter_to(
+      std::int32_t* mem,
+      simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const& index) const {
+    _mm256_mask_i32scatter_epi32(mem, static_cast<__mmask8>(m_mask),
+                                 static_cast<__m256i>(index),
+                                 static_cast<__m256i>(m_value), 4);
```

Comment:
Should it be 4 -> 8 ?

Reply:
Actually why is the scale argument not `1` everywhere?

Reply:
Looking at the pseudo-code for `_mm256_mask_i32scatter_epi32` from [intel intrinsics](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_mask_i32scatter_epi32&ig_expand=3781):
```
addr := base_addr + SignExtend64(vindex[m+31:m]) * ZeroExtend64(scale) * 8
MEM[addr+31:addr] := a[i+31:i]
```
`scale` is really used to represent the byte size of type to determine the stride in bits.

In this case, because this simd type is for `int32_t`, we would need to use 4 for `scale` to make it 32 (bit) * index to get the offset.